### DISCUSSION TITLE
Rendering a PDF on ExpressJS (Saved to file)

### DISCUSSION THREAD

**Discussion by machinsk** [diegomura/react-pdf#1705] [Q&A] [answered]
Hey,

I'm trying to render a PDF on my expressJS backend, and just save it to file. I go into detail in this [stack overflow question](https://stackoverflow.com/questions/70944690/using-react-pdf-renderer-on-an-express-js-server?noredirect=1). But I think I don't have the version of Node that express uses configured correctly to read the ES Modules that the react rendering uses... the server will import my document and the compiler dies on the first sight of a `<` in a `<View>` tag. Any suggestions?

**Comment by machinsk** [comment]
I may have found my [answer](https://github.com/diegomura/react-pdf/discussions/1656)

**Reply by machinsk** [reply]
What is the recommended babel setup and what are babel's dependencies?

**Reply by machinsk** [reply]
I followed these [instructions](https://github.com/diegomura/react-pdf/pull/400/files) but I'm still getting the same error.

```
ReactPDF.render(<MaterialsList />, `${__dirname}/trial.pdf`);
[0]                     ^
[0] 
[0] SyntaxError: Unexpected token '<'
[0]     at Loader.moduleStrategy (internal/modules/esm/translators.js:145:18)
[0]     at async link (internal/modules/esm/module_job.js:63:21)
```

**Reply by machinsk** [reply]
```
import express from 'express';
import cors from 'cors';
import React from 'react';
import ReactPDF from '@react-pdf/renderer';
import MaterialsList from './materials';

const app = express();
const port = 5000;

app.use(cors());

app.use(express.urlencoded());
app.use(express.json());

app.listen(port, function () {
    console.log(`server running on http://localhost:${port}`);
})

app.post('/sendmaterials', cors(), function (req, res) {
    ReactPDF.render(<MaterialsList />, `${__dirname}/trial.pdf`);
})
```

**Reply by machinsk** [reply]
Even starting a new node project with the example code from react-pdf/renderer README gives the same error, including having followed the hidden instructions above....

**Comment by machinsk** [comment, marked as answer]
https://github.com/machinsk/react-pdf-renderer-server-example

**Reply by jeetiss** [reply]
I fixed your example, hope it helps

**Reply by machinsk** [reply]
@jeetiss Thank you for the help. It works! It took me a while to realize I had to run it with `yarn compile` derr..

Thank you very much!!

**Reply by AgDyna** [reply]
> I fixed your example, hope it helps

I am having the same issue, I did what you said,  but the error is still the same.

**Comment by 921011linh** [comment]
Giúp
